pkg/sentry/fs/fdpipe/pipe_opener_test.go

```diff
@@ -421,6 +421,7 @@ func TestPipeHangup(t *testing.T) {
 			hangupSelf: true,
 		},
 	} {
+		test := test // re-create a local copy to avoid data race in the goroutine below
```

Comment:
The goroutine accesses test.flags.Read which is set only once before starting this goroutine. Where is a race?

Reply:
The for loop starts from 424 and the test is an iterator variable that should be captured in the for loop or use a duplicate copy (current fix) for it take effect. 

For example: https://play.golang.org/p/w-0CoJFcE5U

Reply:
Now I see what you mean. I think it is better to pass "test" as an argument in such cases: https://play.golang.org/p/P7acpFBX7on. But I still think that we don't have a race condition here, because this goroutine and the main thread are synchronized via fdchan. This guarantee that "test" isn't changed while the goroutine is running. Have I missed somthing?
https://play.golang.org/p/_gdMGwf7EQ-

Reply:
@avagin sure, I think passing it as argument makes more sense, however test is not a named type here, so it can be passed as interface type: 

```
go func(test interface{}) {
....
}(test)
```
I think we can label it as a (deterministic) race condition since the goroutines race with the main thread over the variable test which the main thread writes with the last (iteration) value before all the goroutines read it.

Reply:
I think there's some confusion: the goroutine is guaranteed not to touch any local variables past line 471, and test is immutable (read-only) for the duration of its existence.

Reply:
@amscanne the issue here is not regarding the data mutability, but that the test within the goroutine is will be the last iterator value unless its captured in the goroutine explicitly. 
For example: https://play.golang.org/p/w-0CoJFcE5U

Reply:
@gaurav1086  your example is wrong.  Pls look at https://play.golang.org/p/_gdMGwf7EQ-, I have already posted it here.

Reply:
@avagin yes my bad, thanks for pointing out.